app/components/autocomplete/date_suggestion/index.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+
+import {getMatchTermForDateMention} from 'app/selectors/autocomplete';
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import DateSuggestion from './date_suggestion';
+
+function mapStateToProps(state, ownProps) {
+    const {cursorPosition, value} = ownProps;
+
+    const newValue = value.substring(0, cursorPosition);
+    const matchTerm = getMatchTermForDateMention(newValue);
+
+    return {
+        matchTerm,
+        theme: getTheme(state),
+    };
+}
+
+function mapDispatchToProps(dispatch) {
```

Comment:
Since we're not using it, you can leave out the `mapDispatchToProps` completely here and just export
```javascript
export default connect(mapStateToProps)(DateSuggestion);
```

Reply:
good point